src/infinite-list/index.tsx

```diff
@@ -0,0 +1,121 @@
+// TODO: Make this a common component for all horizontal lists in this lib
+import React, {forwardRef, useCallback, useMemo, useRef} from 'react';
+import {ScrollViewProps} from 'react-native';
+import {DataProvider, LayoutProvider, RecyclerListView, RecyclerListViewProps} from 'recyclerlistview';
+import inRange from 'lodash/inRange';
+
+import constants from '../commons/constants';
+
+const dataProviderMaker = (items: string[]) => new DataProvider((item1, item2) => item1 !== item2).cloneWithRows(items);
+
+export interface InfiniteListProps
+  extends Omit<RecyclerListViewProps, 'dataProvider' | 'layoutProvider' | 'rowRenderer'> {
+  data: any[];
+  renderItem: RecyclerListViewProps['rowRenderer'];
+  pageWidth?: number;
+  pageHeight?: number;
+  onPageChange?: (pageIndex: number, prevPageIndex: number) => void;
+  onReachEdge?: (pageIndex: number) => void;
+  onReachNearEdge?: (pageIndex: number) => void;
+  nearEdgeThreshold?: number;
```

Comment:
`onReachNearEdgeThreshold` is a better name

Reply:
Done